norfair/metrics.py

```diff
@@ -12,19 +12,18 @@ class InformationFile:
     def __init__(self, file_path):
         with open(file_path, "r") as myfile:
             self.file = myfile.read()
+        self.file = self.file.splitlines()
 
     def search(self, variable_name):
-        index_position_on_this_document = self.file.find(variable_name)
-        index_position_on_this_document = index_position_on_this_document + len(
-            variable_name
-        )
-        while not self.file[index_position_on_this_document].isdigit():
-            index_position_on_this_document += 1
-        value_string = ""
-        while self.file[index_position_on_this_document].isdigit():
-            value_string += self.file[index_position_on_this_document]
-            index_position_on_this_document += 1
-        return int(value_string)
+
+        line = 0
+        while self.file[line][0 : len(variable_name)] != variable_name:
+            line += 1
```

Comment:
Add a catch to the exception that will get raised if we don't find the variable_name with a descriptive error message. As it is currently I think we'll raise a generic index error as the `line` variable will eventually be out of bounds of we don't find the variable_name.

Reply:
I would actually write this class like this:

```python
  class InformationFile:
      def __init__(self, file_path):
          self.path = file_path
          with open(file_path, "r") as myfile:
              self.file = myfile.read()
          self.lines = self.file.splitlines()

      def search(self, variable_name):
          for line in self.lines:
              if line[: len(variable_name)] == variable_name:
                  result = line[len(variable_name) + 1 :]
                  break
          else:
              raise ValueError("Couldn't find '" + variable_name + "' in " + self.path)
          if result.isdigit():
              return int(result)
          else:
              return result
```
Please contact me when you reach this comment as there are several very cool python things to talk about regarding this. Just as a remind for myself those things are ([Exception vs if](https://www.geeksforgeeks.org/try-except-vs-if-in-python/), [for else statements](https://book.pythontips.com/en/latest/for_-_else.html), and [string's index method](https://www.programiz.com/python-programming/methods/string/index)).